pymc_marketing/pytensor_utils.py

```diff
@@ -14,20 +14,326 @@
+        else:
+            new_coords[k] = v
+    f2._coords = new_coords  # type: ignore[attr-defined]
+
+    return f2
+
+
+def merge_models(
+    models: list[Model],
+    *,
+    prefixes: list[str] | None = None,
+    merge_on: str | None = None,
+) -> Model:
+    """Merge multiple PyMC models into a single model.
+
+    Parameters
+    ----------
+    models : list of pm.Model
+        List of models to merge.
+    prefixes : list of str or None
+        List of prefixes for each model. If None, will auto-generate as 'model1', 'model2', ...
+    merge_on : str or None
+        Variable name to merge on (shared across all models) - this variable will NOT be prefixed.
+
+    Returns
+    -------
+    pm.Model
+        Merged model.
+    """
+    if len(models) < 2:
+        raise ValueError("Need at least 2 models to merge")
+
+    # Auto-generate prefixes if not provided
+    if prefixes is None:
+        prefixes = [f"model{i + 1}" for i in range(len(models))]
+    elif len(prefixes) != len(models):
+        raise ValueError(
+            f"Number of prefixes ({len(prefixes)}) must match number of models ({len(models)})"
+        )
+
+    # Variables to exclude from prefixing
+    exclude_vars = {merge_on} if merge_on else set()
+
+    # Convert all models to FunctionGraphs and apply prefixes
+    fgraphs: list[FunctionGraph] = []
+    for model, prefix in zip(models, prefixes, strict=False):
+        fg, _ = fgraph_from_model(model, inlined_views=True)
+        if prefix is not None:
+            fg = _prefix_model(fg, prefix=prefix, exclude_vars=exclude_vars)
+        fgraphs.append(fg)
+
+    # Handle merge_on logic
+    if merge_on is not None:
+        # Find the merge variable in the first model (unprefixed)
+        first_merge_vars = [out for out in fgraphs[0].outputs if out.name == merge_on]
+        if not first_merge_vars:
+            raise ValueError(f"Variable '{merge_on}' not found in first model")
+        first_merge_var = first_merge_vars[0]
+
+        # Replace the merge variable in all other models with the one from the first model
+        for i in range(1, len(fgraphs)):
+            merge_vars = [out for out in fgraphs[i].outputs if out.name == merge_on]
+            if not merge_vars:
+                raise ValueError(f"Variable '{merge_on}' not found in model {i + 1}")
+            fgraphs[i].replace(merge_vars[0], first_merge_var, import_missing=True)
+
+    # Combine all outputs
+    all_outputs = []
+    for fg in fgraphs:
+        all_outputs.extend(fg.outputs)
+
+    # Create merged FunctionGraph
+    f = FunctionGraph(outputs=all_outputs, clone=False)
+
+    # Merge coordinates from all models
+    merged_coords: dict = {}
+    for fg in fgraphs:
+        merged_coords.update(fg._coords)  # type: ignore[attr-defined]
+    f._coords = merged_coords  # type: ignore[attr-defined]
+
+    return model_from_fgraph(f, mutate_fgraph=True)
+
+
+class BuildMergedModel(OptimizerCompatibleModelWrapper):
```

Comment:
If this is causing circular imports, it should be somewhere else. Why not in the budget optimizer module since it is specific to that?